gradio/chat_interface.py

```diff
@@ -462,10 +465,62 @@ def _delete_conversation(
         self,
         index: int | None,
         saved_conversations: list[list[MessageDict]],
+        saved_feedback_values: dict[str, list[str | None]],
     ):
         if index is not None:
             saved_conversations.pop(index)
-        return None, saved_conversations
+            saved_feedback_values.pop(str(index), [])
+        return None, saved_conversations, saved_feedback_values
+
+    def _flag_message(
+        self,
+        conversation: list[MessageDict],
+        conversation_id: int,
+        feedback_values: dict[str, list[str | None]],
+        like_data: LikeData,
+    ) -> dict[str, list[str | None]]:
+        assistant_indices = [
+            i for i, msg in enumerate(conversation) if msg["role"] == "assistant"
+        ]
+        assistant_index = assistant_indices.index(like_data.index)  # type: ignore
+        value = (
+            "Like"
+            if like_data.liked is True
+            else "Dislike"
+            if like_data.liked is False
+            else like_data.liked
+        )
+        feedback_value = feedback_values.get(str(conversation_id), [])
+        if len(feedback_value) <= assistant_index:
+            while len(feedback_value) <= assistant_index:
+                feedback_value.append(None)
+        feedback_value[assistant_index] = value
+        feedback_values[str(conversation_id)] = feedback_value
```

Comment:
`orjson` requires dictionaries to have `str` keys, otherwise serialization fails and the gradio app crashes.